agent/agent_test.go

```diff
@@ -170,6 +170,7 @@ func (s *AgentSuite) TestTaskWithoutSecret() {
 	s.NoError(err)
 	s.Require().NotNil(ntr)
 	s.Equal(false, ntr.shouldExit)
+	s.Equal(true, ntr.noTaskToRun)
```

Comment:
nit: rather than `s.Equal` you can just use `s.False` above this and `s.True` here.